crates/subspace-networking/src/utils.rs

```diff
@@ -95,3 +97,150 @@ pub(crate) fn convert_multiaddresses(addresses: Vec<Multiaddr>) -> Vec<PeerAddre
         })
         .collect()
 }
+
+// Semaphore like implementation that allows both shrinking and expanding
```

Comment:
Data structures and fields in idiomatic Rust should use `///` comments so they are picked up by both compiler and IDE (which, to be fair, is also compiler).